future/transaction.go

```diff
@@ -0,0 +1,530 @@
+package future
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// MinTxnFee is v5 consensus params, in microAlgos
+const MinTxnFee = transaction.MinTxnFee
+
+// MakePaymentTxn constructs a payment transaction using the passed parameters.
+// `from` and `to` addresses should be checksummed, human-readable addresses
+// fee is fee per byte as received from algod SuggestedFee API call
+func MakePaymentTxn(from, to string, amount uint64, note []byte, closeRemainderTo string, params types.SuggestedParams) (types.Transaction, error) {
+	// Decode from address
+	fromAddr, err := types.DecodeAddress(from)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	// Decode to address
+	toAddr, err := types.DecodeAddress(to)
+	if err != nil {
+		return types.Transaction{}, err
+	}
+
+	// Decode the CloseRemainderTo address, if present
+	var closeRemainderToAddr types.Address
+	if closeRemainderTo != "" {
+		closeRemainderToAddr, err = types.DecodeAddress(closeRemainderTo)
+		if err != nil {
+			return types.Transaction{}, err
+		}
+	}
+
+	if len(params.GenesisHash) == 0 {
+		return types.Transaction{}, fmt.Errorf("payment transaction must contain a genesisHash")
+	}
+
+	var gh types.Digest
+	copy(gh[:], params.GenesisHash)
+
+	// Build the transaction
+	tx := types.Transaction{
+		Type: types.PaymentTx,
+		Header: types.Header{
+			Sender:      fromAddr,
+			Fee:         params.Fee,
+			FirstValid:  params.FirstRoundValid,
+			LastValid:   params.LastRoundValid,
+			Note:        note,
+			GenesisID:   params.GenesisID,
+			GenesisHash: gh,
+		},
+		PaymentTxnFields: types.PaymentTxnFields{
+			Receiver:         toAddr,
+			Amount:           types.MicroAlgos(amount),
+			CloseRemainderTo: closeRemainderToAddr,
+		},
+	}
+
+	// Update fee
+	if !params.FlatFee {
+		eSize, err := transaction.EstimateSize(tx)
+		if err != nil {
+			return types.Transaction{}, err
+		}
+		tx.Fee = types.MicroAlgos(eSize * uint64(params.Fee))
+	}
+
+	if tx.Fee < MinTxnFee {
+		tx.Fee = MinTxnFee
+	}
```

Comment:
There was an idea to update the fee while signing the transaction, did the python SDK do that?

Reply:
@winder As far as I can tell, no, changing signing to be the step that updates the fee isn't in the scope of this ticket.

Reply:
that seems fine